warehouse/forklift/legacy.py

```diff
@@ -362,6 +366,92 @@ def _is_duplicate_file(db_session, filename, hashes):
     return None
 
 
+def _process_attestations(request, metrics, artifact_path: Path):
+    """
+    Process any attestations included in a file upload request
+
+    Attestations, if present, will be parsed and verified against the uploaded
+    artifact. Attestations are only allowed when uploading via a Trusted
+    Publisher, because a Trusted Publisher provides the identity that will be
+    used to verify the attestations.
+    Currently, only GitHub Actions Trusted Publishers are supported, and
+    attestations are discarded after verification.
+    """
+
+    # Check that if the file was uploaded with attestations, verification
+    # passes
+    if "attestations" in request.POST:
```

Comment:
Nitpick: I think this check should be outside of this helper function, to avoid having the entire body be in a nested block 🙂